app/services/company.js

```diff
@@ -3,30 +3,93 @@ const path = require('path');
 const companyRepository = require('../repositories/company');
 const addressRepository = require('../repositories/address');
 const fileRepository = require('../repositories/file');
-const fieldOfActivityService = require('./fieldOfActivity');
+const fieldOfActivityRepository = require('../repositories/fieldOfActivity');
 const fileService = require('./file');
 const error = require('../services/error');
 const CrudService = require('./crudService');
 const companyFilesFolder = require('../../config.js').companyFilesFolder;
+const ORIGIN = require('../../config.js').ORIGIN;
 
 class CompanyService extends CrudService {
+  async get(id) {
+    let found = await companyRepository.get(id);
+    if (!found) {
+      error.throwNotFoundError('Company not found.');
+    }
+
+    const { fieldOfActivityId, ...company } = found;
+    const fieldOfActivity =
+      await fieldOfActivityRepository.get(fieldOfActivityId);
+    company.business = fieldOfActivity.name;
+
+    const companyAddresses = await addressRepository.getAllByProps({
+      companyId: id,
+    });
+    if (companyAddresses && companyAddresses.length) {
+      const {
+        streetName,
+        streetNumber,
+        apartmentNumber,
+        city,
+        country,
+      } = companyAddresses[0];
+      company.address = `${streetName} ${streetNumber}, ap. ${apartmentNumber}, ${city}, ${country}`;
+    }
+
+    const companyFiles = await fileRepository.getAllByProps({
+      entityId: id,
+      entityType: 'company',
+    });
+    company.images = companyFiles.map(file => ({
+      name: file.originalFileName,
+      url: `${ORIGIN}companies/${file.originalFileName}`,
+    }));
+
+    return company;
+  }
+
   async create(company, files) {
     if (company.id !== undefined) {
       error.throwValidationError('Invalid company format.');
     }
 
-    const companyExists = await companyRepository.getByPropsNonParanoid({ name: company.name });
+    const companyExists = await companyRepository.getByPropsNonParanoid({
+      name: company.name,
+    });
     if (companyExists) {
       error.throwValidationError('Company already exists.');
     }
-
-    if (company.fieldOfActivityId !== undefined) {
-      const fieldExists = await fieldOfActivityService.get(company.fieldOfActivityId);
+    const { business, address, ...newCompany } = company;
+    if (business !== undefined) {
```

Comment:
This validation logic is taking too much space in this service. Please create another class that handles the validation.